lib/ec2imgutils/ec2deprecateimg.py

```diff
@@ -145,10 +151,34 @@ def _get_all_type_match_images(self, filter_replacement_image=None):
                 if launch_attributes:
                     launch_permission = launch_attributes[0].get('Group', None)
                 if launch_permission == 'all':
-                    images.append(image)
-                continue
-            images.append(image)
+                    should_append_public_only = True
+                else:
+                    # cond specified and image not matching, moving on
+                    continue
 
+            # Append the image once to the list if proceeds
```

Comment:
??? Maybe "Append the image once to the list of images to  be processed" ?

Reply:
my broken english :S